In [3]:
import pandas as pd

trade = pd.read_parquet("../data/processed/clean_trade_data.parquet")
production = pd.read_parquet("../data/processed/clean_production_data.parquet")

In [ ]:
trade.info()

In [13]:
#trade["element"].value_counts(dropna=False)
#trade.loc[trade["value"].isna(), "element"].value_counts(dropna=False)
#trade.loc[trade["value"].notna(), "element"].value_counts(dropna=False)
#trade[trade["value"].isna()].head(10)
trade.loc[trade["value"].isna(), [
    "reporter_countries",
    "partner_countries",
    "item",
    "year",
    "element"
]].head(20)

,reporter_countries,partner_countries,item,year,element
0,Kazakhstan,Austria,Barley,2018,Export quantity
1,Kazakhstan,Austria,Barley,2019,Export quantity
2,Kazakhstan,Austria,Barley,2020,Export quantity
3,Kazakhstan,Austria,Barley,2021,Export quantity
4,Kazakhstan,Austria,Barley,2022,Export quantity
5,Kazakhstan,Austria,Barley,2023,Export quantity
6,Kazakhstan,Austria,Barley,2024,Export quantity
7,Kazakhstan,Austria,Barley,2018,Export value
8,Kazakhstan,Austria,Barley,2019,Export value
9,Kazakhstan,Austria,Barley,2020,Export value


In [21]:
production.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 175 entries, 0 to 174
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   domain_code       175 non-null    object 
 1   domain            175 non-null    object 
 2   area_code_m49     175 non-null    int64  
 3   area              175 non-null    object 
 4   element_code      175 non-null    int64  
 5   element           175 non-null    object 
 6   item_code_cpc     175 non-null    float64
 7   item              175 non-null    object 
 8   year_code         175 non-null    int64  
 9   year              175 non-null    Int64  
 10  unit              159 non-null    object 
 11  value             159 non-null    float64
 12  flag              159 non-null    object 
 13  flag_description  159 non-null    object 
dtypes: Int64(1), float64(2), int64(3), object(8)
memory usage: 19.4+ KB


In [23]:
#production["element"].value_counts(dropna=False)
#production.loc[production["value"].isna(), "element"].value_counts(dropna=False)
#production.loc[production["value"].notna(), "element"].value_counts(dropna=False)
production[production["value"].isna()].head(20)

,domain_code,domain,area_code_m49,area,element_code,element,item_code_cpc,item,year_code,year,unit,value,flag,flag_description
105,QCL,Crops and livestock products,398,Kazakhstan,5423,Extraction Rate,21641.01,"Rapeseed or canola oil, crude",2018,2018,None,NaN,None,None
107,QCL,Crops and livestock products,398,Kazakhstan,5423,Extraction Rate,21641.01,"Rapeseed or canola oil, crude",2019,2019,None,NaN,None,None
109,QCL,Crops and livestock products,398,Kazakhstan,5423,Extraction Rate,21641.01,"Rapeseed or canola oil, crude",2020,2020,None,NaN,None,None
111,QCL,Crops and livestock products,398,Kazakhstan,5423,Extraction Rate,21641.01,"Rapeseed or canola oil, crude",2021,2021,None,NaN,None,None
113,QCL,Crops and livestock products,398,Kazakhstan,5423,Extraction Rate,21641.01,"Rapeseed or canola oil, crude",2022,2022,None,NaN,None,None
115,QCL,Crops and livestock products,398,Kazakhstan,5423,Extraction Rate,21641.01,"Rapeseed or canola oil, crude",2023,2023,None,NaN,None,None
117,QCL,Crops and livestock products,398,Kazakhstan,5423,Extraction Rate,21641.01,"Rapeseed or canola oil, crude",2024,2024,None,NaN,None,None
118,QCL,Crops and livestock products,398,Kazakhstan,5510,Production,21641.01,"Rapeseed or canola oil, crude",2024,2024,None,NaN,None,None
140,QCL,Crops and livestock products,398,Kazakhstan,5423,Extraction Rate,21631.01,"Sunflower-seed oil, crude",2018,2018,None,NaN,None,None
142,QCL,Crops and livestock products,398,Kazakhstan,5423,Extraction Rate,21631.01,"Sunflower-seed oil, crude",2019,2019,None,NaN,None,None


============================================================
Cleaning Summary

Trade dataset
-------------
Original rows: 3528
Duplicate rows removed: 0
Rows with missing value: 3014
Output rows: 3528

Production dataset
------------------
Original rows: 175
Duplicate rows removed: 0
Rows with missing value: 16
Output rows: 175
============================================================